File: mns-core/src/test/java/cz/fb/manaus/core/model/RunnerTest.java
package cz.fb.manaus.core.model;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Test;

import static org.hamcrest.CoreMatchers.containsString;
import static org.junit.Assert.assertThat;

public class RunnerTest {

    public static final ObjectMapper MAPPER = new ObjectMapper();

    public static Runner create(long selectionId, String name, double handicap, int sortPriority) {
        var runner = new Runner();
        runner.setSelectionId(selectionId);
        runner.setName(name);
        runner.setHandicap(handicap);
        runner.setSortPriority(sortPriority);
        return runner;
    }

    @Test
    public void testMarshallJson() throws Exception {
        Runner runner = create(100, "Sparta", 0, 0);
        String json = MAPPER.writeValueAsString(runner);
        assertThat(json, containsString("Sparta"));
    }

}

File: mns-core/src/test/java/cz/fb/manaus/core/model/RunnerTest.kt
package cz.fb.manaus.core.model

import com.fasterxml.jackson.databind.ObjectMapper
import org.hamcrest.CoreMatchers.containsString
import org.junit.Assert.assertThat
import org.junit.Test

class RunnerTest {

    @Test
    fun `json marshall`() {
        val runner = ModelFactory.newRunner(100, "Sparta", 0.0, 0)
        val json = ObjectMapper().writeValueAsString(runner)
        assertThat(json, containsString("Sparta"))
    }
}